docs/source/federation-spec.md

```diff
@@ -309,3 +309,38 @@ extend type User @key(fields: "email") {
 ```
 
 This type extension in the Reviews service extends the `User` type from the Users service. It extends it for the purpose of adding a new field called `reviews`, which returns a list of `Review`s.
+
+### `@link`
+
+```graphql
+scalar link__Import
+
+enum link__Purpose {
+  """
+  \`SECURITY\` features provide metadata necessary to securely resolve fields.
+  """
+  SECURITY
+
+  """
+  \`EXECUTION\` features provide metadata necessary for operation execution.
+  """
+  EXECUTION
+}
+
+directive @link(
+  url: String, 
+  as: String, 
+  for: link__Purpose, 
+  import: [link__Import]
+) repeatable on SCHEMA
+```
+
+The `@link` directive provides a mechanism to link to external schemas, specifying the URL, version, and directives. The presence of a `@link` directive also indicates that the schema is a core schema. 
```

Comment:
```suggestion
The `@link` directive links definitions within the document to external schemas. External schemas are identified by their `url`, which optionally ends with a name and version in the form `name`/v`major`.`minor`.

The presence of a `@link` directive makes a document a [core schema](https://specs.apollo.dev/#def-core-schema).

The `for` argument describes the purpose of a `@link`. Currently accepted values are `SECURITY` or `EXECUTION`. Core schema-aware servers such as Apollo Router and Gateway will refuse to operate on schemas which contain `@link`s to unsupported specs which are `for: SECURITY` or `for: EXECUTION`.

By default, `@link`ed definitions will be namespaced, i.e. `@federation__requires`. The `import` argument lets you import external definitions into your namespace, so they lose the prefix:
```

Reply:
@queerviolet Should that last bit be
> The `as` argument lets you import

I might also word golf that a little
> By default, `@link`ed definitions are namespaced with the name of the schema, e.g. `@federation__requires`. The `as` argument lets you import definitions without the prefix.

Reply:
@queerviolet Should that last bit be
> The `as` argument lets you import

I might also word golf that a little if we're open to it:

> By default, `@link`ed definitions are namespaced with the name of the schema, e.g. `@federation__requires`. The `as` argument lets you import definitions without the prefix.